amd/src/login.js

```diff
@@ -0,0 +1,214 @@
+        } else {
+            // Populate the username field in the password form
+            $options.password.username.val(username);
+            $options.password.form.submit();
+        }
+    }
+
+    function handlePasswordSubmission($event) {
+        $event.preventDefault();
+        var $options = $event.data;
+        var username = $options.password.username.val();
+
+        queryWebservice(username)[0].done(function(response) {
+            var currentURL = new URL(M.cfg.wwwroot).hostname;
+            if(response.domain !== currentURL) {
+                handleDomainRedirect(response, username);
+            } else {
+                $options.password.form.submit();
+            }
+        });
+    }
+
+    function setRememberme($options) {
+        var $remFromUserForm = $options.username.rememberme;
+        var $remFromPassForm = $options.password.rememberme;
+        if ($remFromUserForm.prop('checked')) {
+            $remFromPassForm.val("1");
+        }
+    }
+
+    function setReturnurl($options) {
+        var retFromUserFormVal = $options.username.returnurl.val();
+        var $retFromPassForm = $options.password.returnurl;
+        if (retFromUserFormVal) {
+            $retFromPassForm.val(retFromUserFormVal);
+        }
+    }
+
+    function checkDomain(username, $options) {
+        var result = false;
+        queryWebservice(username)[0].done(function(response) {
+            result = handleDomainRedirect(response, username, $options);
+        }).fail(function() {
+            get_string_and_notify('danger', 'invalid_user');
+            result = false;
+        });
+        return result;
+    }
+
+    function queryWebservice(username){
+        return ajax.call([
+            {
+                methodname: WEB_SERVICE_METHOD_NAME,
+                args: {
+                    username: username
+                }
+            }
+        ]);
+    }
+
+    function handleDomainRedirect(response, username, $options) {
+        var currentURL = new URL(M.cfg.wwwroot);
+        if (response.email === null) {
+            // flag non-existent user error
+            get_string_and_notify('warning', 'non_existent_user');
+            return false;
+        } else if (response.domain !== currentURL.hostname) {
+            // Redirect to recorded user default domain
+            window.location = window.location.protocol + '//' + response.domain + currentURL.pathname +
+                '/local/signin/index.php?username=' + username;
+            return false;
+        } else {
+            $(defaults.form.alert).alert('close');
+            // Stay on page & flip over to password form
+            // Populate username field on password form
+            $options.password.username.val(username);
+            // Toggle the password form
+            $options.password.changeuser.click();
+            $options.password.submit.on('click', $options.dom, $options.password.validation);
```

Comment:
This event should be wired up in `init()` -- you can still add events to nodes that aren't yet visible.